topo/node/alpine/alpine.go

```diff
@@ -0,0 +1,176 @@
+			Command:         pb.Config.Command,
+			Args:            pb.Config.Args,
+			Env:             node.ToEnvVar(pb.Config.Env),
+			Resources:       node.ToResourceRequirements(pb.Constraints),
+			ImagePullPolicy: "IfNotPresent",
+			SecurityContext: &corev1.SecurityContext{
+				Privileged: pointer.Bool(true),
+			},
+		},
+	}
+
+	if vendorData := pb.Config.GetVendorData(); vendorData != nil {
+		alpineDpConfig := &apb.AlpineDataplaneConfig{}
+		if err := vendorData.UnmarshalTo(alpineDpConfig); err != nil {
+			return err
+		}
+		dataplaneName := alpineDpConfig.Name
+		dataplaneImage := alpineDpConfig.Image
+		dataplaneCommand := alpineDpConfig.Command
+		dataplaneArgs := alpineDpConfig.Args
+
+		alpineContainers = append(alpineContainers,
+			corev1.Container{
+				Name:            dataplaneName,
+				Image:           dataplaneImage,
+				Command:         dataplaneCommand,
+				Args:            dataplaneArgs,
+				Env:             node.ToEnvVar(pb.Config.Env),
+				Resources:       node.ToResourceRequirements(pb.Constraints),
+				ImagePullPolicy: "IfNotPresent",
+				SecurityContext: &corev1.SecurityContext{
+					Privileged: pointer.Bool(true),
+				},
+			},
+		)
+	}
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: pb.Name,
+			Labels: map[string]string{
+				"app":  pb.Name,
+				"topo": n.Namespace,
+			},
+		},
+		Spec: corev1.PodSpec{
+			InitContainers: []corev1.Container{{
+				Name:  fmt.Sprintf("init-%s", pb.Name),
+				Image: initContainerImage,
+				Args: []string{
+					fmt.Sprintf("%d", len(n.Proto.Interfaces)+1),
+					fmt.Sprintf("%d", pb.Config.Sleep),
+				},
+				ImagePullPolicy: "IfNotPresent",
+			}},
+			Containers: alpineContainers,
+			TerminationGracePeriodSeconds: pointer.Int64(0),
+			NodeSelector:                  map[string]string{},
+			Affinity: &corev1.Affinity{
+				PodAntiAffinity: &corev1.PodAntiAffinity{
+					PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{{
+						Weight: 100,
+						PodAffinityTerm: corev1.PodAffinityTerm{
+							LabelSelector: &metav1.LabelSelector{
+								MatchExpressions: []metav1.LabelSelectorRequirement{{
+									Key:      "topo",
+									Operator: "In",
+									Values:   []string{pb.Name},
+								}},
+							},
+							TopologyKey: "kubernetes.io/hostname",
+						},
+					}},
+				},
+			},
+		},
+	}
+	sPod, err := n.KubeClient.CoreV1().Pods(n.Namespace).Create(ctx, pod, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+	log.Infof("Pod created:\n%+v\n", sPod)
+	return nil
+}
+
+func defaults(pb *tpb.Node) *tpb.Node {
+	if pb.Config == nil {
+		pb.Config = &tpb.Config{}
+	}
+	if len(pb.GetConfig().GetCommand()) == 0 {
+		pb.Config.Command = []string{"/bin/sh", "-c", "sleep 2000000000000"}
+	}
+	if pb.Config.EntryCommand == "" {
+		pb.Config.EntryCommand = fmt.Sprintf("kubectl exec -it %s -- sh", pb.Name)
+	}
+	if pb.Config.Image == "" {
+		pb.Config.Image = "sonic-vs:latest"
```

Comment:
is this image public? if not I would recommend the default to be `alpine:latest`

unless you consider alpine to be this entire bundle and sonic is the name of the control plane, in which case a default of `sonic:latest` would work too

Reply:
Ack, using alpine:latest for now